app/controllers/concerns/statistics.rb

```diff
@@ -0,0 +1,32 @@
+module Statistics
+  extend ActiveSupport::Concern
+
+  def dates_and_ranges
+    @start = DateTime.parse('2017-01-01').in_time_zone
+    @one_month = 1.month.ago..Time.zone.now
+    @one_year = 1.year.ago..Time.zone.now
+    @from_start = @start..Time.zone.now
+  end
+
+  def period
+    dates_and_ranges
+    return @one_month if params[:period] == 'month'
+    if params[:period] == 'year'
+      return 1.year.ago < @start ? @from_start : @one_year
+    end
+    return @from_start if params[:period] == 'total'
+    @one_month
+  end
+
+  def day_range
+    ((period.last - period.first) / 1.day).ceil
+  end
+
+  def group_by
+    day_range
```

Comment:
@igorkasyanchuk, on development env. we use SQLite DB and it isn't support `:quarter`, so we use it only for production with PostgreSQL, otherwise method returns `:week`.
Line #26 will be fixed.